packages/tasks/src/local-apps.ts

```diff
@@ -0,0 +1,119 @@
+import type { ModelData } from "./model-data";
+import type { PipelineType } from "./pipelines";
+
+/**
+ * Elements configurable by a local app.
+ */
+export type LocalApp = {
+	/**
+	 * Name that appears in buttons
+	 */
+	prettyLabel: string;
+	/**
+	 * Link to get more info about a local app (website etc)
+	 */
+	docsUrl: string;
+	/**
+	 * main category of app
+	 */
+	mainTask: PipelineType;
+	/**
+	 * Whether to display a pill "macOS-only"
+	 */
+	macOSOnly?: boolean;
+
+	comingSoon?: boolean;
+	/**
+	 * IMPORTANT: function to figure out whether to display the button on a model page's main "Use this model" dropdown.
+	 */
+	displayOnModelPage: (model: ModelData) => boolean;
+} & (
+	| {
+			/**
+			 * If the app supports deeplink, URL to open.
+			 */
+			deeplink: (model: ModelData) => URL;
+	  }
+	| {
+			/**
+			 * And if not (mostly llama.cpp), snippet to copy/paste in your terminal
+			 */
+			snippet: (model: ModelData) => string;
+	  }
+);
+
+function isGgufModel(model: ModelData) {
+	return model.tags.includes("gguf");
+}
+
+const snippetLlamacpp = (model: ModelData): string => {
+	return `llama-cpp \
+	--hf-repo ${model.id} \
+	-m file.gguf \
+	-p "I believe the meaning of life is " -n 128`;
+};
+
+/**
+ * Add your new local app here.
+ *
+ * This is open to new suggestions and awesome upcoming apps.
+ *
+ * /!\ IMPORTANT
+ *
+ * If possible, you need to support deeplinks and be as cross-platform as possible.
+ *
+ * Ping the HF team if we can help with anything!
+ */
+export const LOCAL_APPS = {
+	"llama.cpp": {
+		prettyLabel: "llama.cpp",
+		docsUrl: "https://github.com/ggerganov/llama.cpp",
+		mainTask: "text-generation",
+		displayOnModelPage: isGgufModel,
+		snippet: snippetLlamacpp,
+	},
+	lmstudio: {
+		prettyLabel: "LM Studio",
+		docsUrl: "https://lmstudio.ai",
+		mainTask: "text-generation",
+		displayOnModelPage: isGgufModel,
+		deeplink: (model) => new URL(`lmstudio://open_from_hf?model=${model.id}`),
+	},
+	jan: {
+		prettyLabel: "Jan",
+		docsUrl: "https://jan.ai",
+		mainTask: "text-generation",
+		displayOnModelPage: isGgufModel,
```

Comment:
AFAIR Jan works for both GGUFs as well as TRT-LLM too, would be good to add support for that if possible too! 🙏

Reply:
let's do it in a subsequent PR if it works for you @Vaibhavs10

Reply:
Yes! also chatting with Jan folks on this.